### DISCUSSION TITLE
PCAP No data available

### DISCUSSION THREAD

**Discussion by coreyman2000** [Security-Onion-Solutions/securityonion#9495] [Unsupported Versions]
Hi new install from ISO,  when i attempt to pcap from a hunt i get No data available, ive tired using the docker command as well no luck i get 0 items.

i see the /nsm/pcap has a bunch of files, however when i try to hunt pcap its blank

again fresh install, nothing modified 

"
cat config

{
  "Threads": [
    { "PacketsDirectory": "/nsm/pcap", "IndexDirectory": "/nsm/pcapindex", "MaxDirectoryFiles": 30000, "DiskFreePercentage": 10 }
  ]
  , "StenotypePath": "/usr/bin/stenotype"
  , "Interface": "bond0"
  , "Port": 1234
  , "Host": "127.0.0.1"
  , "Flags": ["-v", "--blocks=2048", "--preallocate_file_mb=4096", "--aiops=128", "--uid=stenographer", "--gid=stenographer"]
  , "CertPath": "/etc/stenographer/certs"

"

**Comment by dougburks** [comment]
Have you checked logs for additional clues?
https://docs.securityonion.net/en/2.3/stenographer.html#diagnostic-logging

If you need additional help, please provide more information:

How is your installation configured?

Are you consuming traffic from a tap, span, cloud, or other?

Are you able to provide screenshots of an example log that you are you trying to pivot to pcap?

**Comment by coreyman2000** [comment]
thanks for taking the time to respond 

I am looking at the logs now, will update if there anything relevant however the logs look pretty emtpy 
`sudo docker logs so-steno
Generating CA state
Generating key/cert for 'client'
Generating key/cert for 'server'
Skipping CA state generation, ca_cert.pem already exists
Skipping generation of 'client_127.0.0.1_client' key/cert, client_127.0.0.1_client_cert.pem already exists
Skipping generation of 'server_127.0.0.1' key/cert, server_127.0.0.1_cert.pem already exists
Skipping CA state generation, ca_cert.pem already exists
Skipping generation of 'client_127.0.0.1_client' key/cert, client_127.0.0.1_client_cert.pem already exists
Skipping generation of 'server_127.0.0.1' key/cert, server_127.0.0.1_cert.pem already exists
[root@securityonion admin]# docker volume create portainer_data
`

How is your installation configured?  **standalone , default install**
Are you consuming traffic from a tap, span, cloud, or other? **im using a mirrored port from the WAN port of the router.**

Screenshots
![image](https://user-images.githubusercontent.com/21960367/210368657-8addb653-2545-4db6-a646-35e38a4e8c8f.png)

![image](https://user-images.githubusercontent.com/21960367/210369009-bbf677ce-87eb-4b85-994c-e289a61e3233.png)

Also when created an a new job same resulsts
![image](https://user-images.githubusercontent.com/21960367/210371113-c629864b-51af-4b7c-a827-058a84f33dcc.png)

![image](https://user-images.githubusercontent.com/21960367/210370987-34fa5130-0106-458e-8b8e-cccbc0940f55.png)

![image](https://user-images.githubusercontent.com/21960367/210371263-5ea0111f-0f5c-4aaf-97cd-2155ca950805.png)

**Reply by dougburks** [reply]
> Are you consuming traffic from a tap, span, cloud, or other? **im using a mirrored port from the WAN port of the router.**

Based on this answer and looking at the first screenshot, the alerts appear to be simple Internet scanning.

Here's the rule that generated the alerts:
```
alert ip [92.205.161.70,92.205.164.27,92.21.106.61,92.222.120.150,92.242.38.58,92.246.179.73,92.249.123.55,92.255.238.231,92.255.85.15,92.255.85.16,92.255.85.9,92.33.135.51,92.33.150.226,92.33.187.223,92.33.248.84,92.39.92.26,92.44.114.243,92.51.220.122,92.55.100.136,92.63.196.130,92.63.196.4,92.63.196.69,92.63.197.100,92.63.197.131,92.63.197.61,92.63.197.74,92.63.197.94,92.63.206.50,92.68.41.253,92.84.225.50,93.108.12.140,93.118.109.108,93.118.121.76,93.126.21.27,93.126.72.9,93.147.129.191,93.157.172.89,93.158.66.10,93.159.235.155,93.171.143.110,93.174.164.44,93.174.95.106,93.175.205.158,93.176.174.28,93.176.190.141,93.176.220.31,93.177.103.227,93.180.134.140,93.185.215.24,93.188.125.1] any -> $HOME_NET any (msg:"ET CINS Active Threat Intelligence Poor Reputation IP group 88"; reference:url,www.cinsscore.com; threshold: type limit, track by_src, seconds 3600, count 1; classtype:misc-attack; sid:2403387; rev:80028; metadata:affected_product Any, attack_target Any, deployment Perimeter, tag CINS, signature_severity Major, created_at 2013_10_08, updated_at 2023_01_03;)
```
This is a simple rule that would fire solely based on IP address even if no real data was transferred.

This scanning is most likely being blocked by your firewall so when you pivot to packet capture there is no real data to show.

My next step would be to go to Dashboards and select the HTTP dashboard. Then see if you see the HTTP transactions that you expect to see on your WAN port. If so, then pivot to pcap on one of those HTTP transactions and see if it shows the full TCP stream as expected.

**Comment by coreyman2000** [comment]
this traffic is before the physical firewall, so i would except it to capture it,  unless do you mean the SO  (centos )local firewall blocks it?

so here is the HTTP dashboard and i do see traffic 
![image](https://user-images.githubusercontent.com/21960367/210584646-5d80f0a7-f478-4cc1-bf82-c23f116b58e5.png)
![image](https://user-images.githubusercontent.com/21960367/210584799-876ac60b-1d0d-4beb-bc48-d98f74486ee7.png)

i still get no data , am i doing something wrong?
![image](https://user-images.githubusercontent.com/21960367/210584880-f5cfde1f-3f9f-4e0b-86f4-9eb573c407ef.png)

**Reply by dougburks** [reply]
Are you trying to pivot to pcap immediately after the HTTP transaction was logged? If so, and depending on how much traffic you're monitoring, it may take a few minutes for the pcap to be available. Have you tried pivoting on an HTTP transaction that is from the previous day to see if that behaves differently?

**Comment by spirrello** [comment]
@dougburks I'm running version `2.3.210` and see the same behavior on a fresh install.

**Reply by spirrello** [reply]
@dougburks Side note, I installed my instance in AWS using the AMI from the market place.  I've confirmed the traffic mirror session is working and can see captures in the `/nsm/pcapindex/` and `/nsm/pcap` directories.

**Reply by dougburks** [reply]
@spirrello Please review https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and in particular this part:

> **Start a new discussion instead of replying to somebody else's discussion**
Please search to see if you can find similar discussions that may help you. However, in order to avoid confusion, please do NOT reply to somebody else's discussion with your own issue. Instead, please start a new discussion and in that new discussion you can provide a hyperlink to the related discussion.
